### PR TITLE
test(commands): stores gpg keyring in a temporary directory

### PR DESCRIPTION
**Pull request by bhelgs**
## Description
The unit tests were creating a bunch of keys (run tests a few times and see `gpg --list-keys`).

This solution creates a temporary `GNUPGHOME`. Notably gpg-agent requires a socket path of less 104 characters otherwise the gpg command will fail.

I am not confident this is the best approach but it is probably the easiest fix for the moment.
Is this a portable solution?

Also, I reduced the key length down to a debug level key for speed and security reasons.

## Checklist

- [x] Add test cases to all the changes you introduce
- [x] Run `./scripts/format` and `./scripts/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Expected behavior
no more added keys.

## Steps to Test This Pull Request
1. ran tests.
2. observed  `gpg --list-keys`

### CODE REVIEW

**Comment by gpongelli** [comment]
I saw the tests create a bunch of keys, but it is a drawback for package’s developer and not for final user.

Why don’t use a [TemporaryDirectory](https://docs.python.org/3/library/tempfile.html) instead of home’s subdir?

Anyway, good improvement !

**Comment by bhelgs** [comment]
A TemporaryDirectory was my first thought as well, but is there a way to enforce that the path to the sockets within the directory is shorter than 104 characters?

**Comment by gpongelli** [comment]
> A TemporaryDirectory was my first thought as well, but is there a way to enforce that the path to the sockets within the directory is shorter than 104 characters?

IDK

**Comment by bhelgs** [comment]
I can do a bit of digging in a bit.

**Comment by Lee-W** [comment]
I might try something like [tmp_path](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture), get the length of current path and trim it to 104. Or maybe we could use `/tmp`? 

Overall the PR looks great! Looking forward to merging it!

**Comment by gpongelli** [comment]
> I might try something like [tmp_path](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture), get the length of current path and trim it to 104. Or maybe we could use `/tmp`?
> 
> Overall the PR looks great! Looking forward to merging it!

The [TemporaryDirectory](https://docs.python.org/3/library/tempfile.html) is already under `/tmp` (or whatever tmp folder depending by OS) and it is context managed.

**Comment by bhelgs** [comment]
I was running into issues using `TemporaryDirectory`. 

TemporaryDirectory is sensitive to the [TMPDIR variable, among others](https://docs.python.org/3.4/library/tempfile.html#tempfile.tempdir).

* `$TMPDIR` on OSX:
    * `/var/folders/??/???????????_?_????????????????/T/`
    * Each path is sorted in this way since so that a single directory of the filesystem doesn't get over cluttered or have performance issues.
* Windows:
    * `C:\Users\<username>\AppData\Local\Temp`
    * though gpg-agent may not rely upon socket paths within the Windows implementation.

I thought about just overwriting (or erasing) the one of these environment variables but that gets us involved in an OS specific solution. 

The pytest `tmp_path`/`tmpdir` also runs into similar issues (`/private/var/folders/??/???????????_?_????????????????/T/pytest-of-<username>/pytest-???/`) except the final couple of directories seems to be not customizable. 

I feel there should be a standard solution for a character constrained temporary directory though. Might I be missing something?

**Comment by bhelgs** [comment]
# tl;dr (too long; did not read)
I think we can count on the `TemporaryDirectory` solution being less than 104 characters. I'll give it another try. Thanks for your feedback.

# pytest `tmp_path`/`tmpdir`
The pytest `tmp_path`/`tmpdir` on OSX is (`/private/var/folders/??/???????????_?_????????????????/T/pytest-of-<username>/pytest-???/`). The final couple of directories do not seem to be not customizable. This sets us over the 104 character limit, at least on OSX it did.

# `TemporaryDirectory`:
`TemporaryDirectory` looks like it should work.

`TemporaryDirectory` is sensitive to the [`TMPDIR` variable, among others](https://docs.python.org/3.4/library/tempfile.html#tempfile.tempdir).

* $TMPDIR on OSX:
   * `/var/folders/??/???????????_?_????????????????/T/`
   * Each path is sorted in this way since so that a single directory of the filesystem doesn't get over cluttered or have performance issues.
   * The subdirectory is a relatively short/unique string so I think we are under 104 characters.
* Windows:
   * `C:\Users\<username>\AppData\Local\Temp`
   * path seems short and `gpg-agent` may not rely upon socket paths within the Windows implementation.
* Linux:
   * similar situation to OSX but it is distribution specific. overwriting $TMPDIR is a workaround should the event arise.

**Comment by bhelgs** [comment]
I think this change is good to go now. Thanks for the input everyone!

Any guess as to why the coverage check failed? It seems like it should have passed.

**Review by Woile — APPROVED** [review]
LGTM, @Lee-W if you agree let's merge

**Review by gpongelli — CHANGES_REQUESTED** [review]
Typo error on git config

File: tests/conftest.py [hunk 1]
```diff
@@ -24,34 +25,39 @@ def tmp_commitizen_project(tmp_git_project):
...
+    c = cmd.run(
+        f"gpg --batch --yes --debug-quick-random --passphrase '' --quick-gen-key '{signer} {signer_mail}'"
+    )
+    if c.return_code != 0:
+        raise Exception(f"gpg keygen failed with err: '{c.err}'")
+    key_id = _get_gpg_keyid(signer_mail)
+    assert key_id
+
+    # configure git
+    cmd.run("git config commit.gpgsignq true")
```
[the remark this answers is not in the document]
Comment (Lee-W) [reply tests/conftest.py:57]: according to the example [here](https://docs.gitlab.com/ee/user/project/repository/gpg_signed_commits/#sign-your-git-commits), looks like a typo

**Review by Lee-W — APPROVED** [review]
I'm good with this one. Once the typo has been fixed. I think we're good to merge this one

File: tests/conftest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhelgs) [reply tests/conftest.py]: fixed, thanks!

**Comment by bhelgs** [comment]
@Woile, I think it's good now, thanks!

**Comment by Lee-W** [comment]
many thanks for your contribution! let's merge it!
